framework/src/utils/JsonSyntaxTree.C

```diff
@@ -120,22 +124,30 @@ JsonSyntaxTree::addParameters(const std::string & parent, const std::string & pa
   }
   if (_search != "" && count == 0)
     // no parameters that matched the search string
-    return;
+    return false;
 
   moosecontrib::Json::Value & json = getJson(parent, path, is_type);
+
   if (is_action)
   {
     json[action]["parameters"] = all_params;
-    json[action]["tasks"].append(task_name);
+    json[action]["description"] = params->getClassDescription();
     json[action]["action_path"] = path;
+    auto & fi = json[action]["file_info"];
```

Comment:
Yes, this and the other one are intentional. This always puts a `file_info` in the json, it just might be `null` if there is no information. This allows the parser to not have to check if the key exists. But I can remove it if you like.